backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientDataResolver.java

```diff
@@ -7,51 +7,80 @@
-  }
+import java.util.stream.Collectors;
+import org.dataloader.DataLoader;
+import org.springframework.graphql.data.method.annotation.SchemaMapping;
+import org.springframework.graphql.execution.BatchLoaderRegistry;
+import org.springframework.stereotype.Controller;
+import reactor.core.publisher.Mono;
+
+@Controller
+public class PatientDataResolver implements PersonNameResolver<Person>, InternalIdResolver<Person> {
+
+  public PatientDataResolver(
+      BatchLoaderRegistry registry,
+      TestEventRepository testEventRepository,
+      PhoneNumberRepository phoneNumberRepository) {
+    registry
+        .forTypePair(UUID.class, TestEvent.class)
+        .withName("patientLastTestLoader")
+        .registerMappedBatchLoader(
+            (patientIds, batchLoaderEnvironment) -> {
+              Map<UUID, TestEvent> found =
+                  testEventRepository.findLastTestsByPatient(patientIds).stream()
+                      .collect(Collectors.toMap(TestEvent::getPatientInternalID, s -> s));
+              return Mono.just(found);
+            });
+
+    Class<List<PhoneNumber>> phoneNumberListClazz = (Class) List.class;
+    registry
+        .forTypePair(UUID.class, phoneNumberListClazz)
+        .withName("patientPhoneNumbersLoader")
+        .registerMappedBatchLoader(
+            (patientIds, batchLoaderEnvironment) -> {
+              Map<UUID, List<PhoneNumber>> found =
+                  phoneNumberRepository.findAllByPersonInternalIdIn(patientIds).stream()
+                      .flatMap(List::stream)
+                      .collect(Collectors.groupingBy(PhoneNumber::getPersonInternalID));
 
-  public CompletableFuture<TestEvent> getLastTest(Person person, DataFetchingEnvironment dfe) {
-    return _patientLastTestDataLoader.load(person.getInternalId(), dfe);
+              return Mono.just(found);
+            });
   }
 
-  public ApiFacility getFacility(Person p) {
-    Facility f = p.getFacility();
-    return f == null ? null : new ApiFacility(f);
+  @SchemaMapping(typeName = "Patient", field = "lastTest")
+  public CompletableFuture<TestEvent> getLastTest(
+      Person person, DataLoader<UUID, TestEvent> patientLastTestLoader) {
+    return patientLastTestLoader.load(person.getInternalId());
   }
 
+  @SchemaMapping(typeName = "Patient", field = "phoneNumbers")
   public CompletableFuture<List<PhoneNumber>> getPhoneNumbers(
-      Person person, DataFetchingEnvironment dfe) {
-    return _patientPhoneNumbersDataLoader.load(person.getInternalId(), dfe);
+      Person person, DataLoader<UUID, List<PhoneNumber>> patientPhoneNumbersLoader) {
+    return patientPhoneNumbersLoader.load(person.getInternalId());
+  }
+
+  @SchemaMapping(typeName = "Patient", field = "facility")
+  public ApiFacility facility(Person patient) {
+    Facility f = patient.getFacility();
+    return f == null ? null : new ApiFacility(f);
   }
 
-  public CompletableFuture<String> getTelephone(Person person, DataFetchingEnvironment dfe) {
```

Comment:
This has been removed, and currently its been switched to Eager loading
previously it was using the _patientPrimaryPhoneDataLoader
which the source code is located [here](https://github.com/CDCgov/prime-simplereport/blob/main/backend/src/main/java/gov/cdc/usds/simplereport/service/dataloader/PatientPrimaryPhoneDataLoader.java#L14)

The data loader was doing something very unintelligent, notice how the data loader was [using the getPrimaryPhone to load the phone number](https://github.com/CDCgov/prime-simplereport/blob/main/backend/src/main/java/gov/cdc/usds/simplereport/service/dataloader/PatientPrimaryPhoneDataLoader.java#L30), then it was preceeding to get the PhoneNumber phone id, then it was grabbing all these PhoneNumbers internal ids, and then [bulk loading them again from the db](https://github.com/CDCgov/prime-simplereport/blob/main/backend/src/main/java/gov/cdc/usds/simplereport/service/dataloader/PatientPrimaryPhoneDataLoader.java#L41)
so it was kinda doing more work than needed and actively hurting performance.